### PR TITLE
Add Compatibility to crawlers render pages

### PR DESCRIPTION
**Pull request by royalterra**
Add compatibility to crawlers render pages. No extra configuration need by user. Changes already tested

### CODE REVIEW

**Comment by royalterra** [comment]
@redmatrix I not believe the previous pull is round works nearly perfect. The issue is how some Crawlers like to start follow Urls . Because I can't see how works Yahoo or Archive.org I use webmaster tools with Google give me the possibility to force a crawler and not need to wait and also I CAN render to force a cache page.
The domains google have not render, render means" Google or others  is getting a cache page " Metatags work perfect just display in the Crawler results what ever I setup in .httconfig.php .  And the results are more I expected forcing example blablanet.fr to be show in France with French metatags have make more users of any other of my hubzilla domains in only one Month . The other domains like blablanet.com have take 3 months I believe now for get the seam number of users and blablanet.fr less of one Month . That show Metatags accomplish what we need for get more users. And make Hubzilla bigger Network
 
Now after I see some issues with the urls in the chinese domain because for any reason the search engines consider is more important of one domain.com  I follow the crawler and will make a paste bin here of google search engine.

For some reason when the search engine try to follow all the index of home page and make a cache of all the LINKS the search engine get Redirects in the urls by example home page need to be "/" become  in hubzilla "/&JS=1" and like that all the Hubzilla Url That is not a issue of Metatags is a issue of Hubzilla code. I see that url also too in Firefox in android after open again the cache page in the browser. Now metatags work with links like home is / and register, login etc but when a search engine request a url ending with the variable of course not work. Is for that after I observe the issue I add a extra configuration with the ending variable for cover any possibility's. And I test and now with a simple crawl or with a render the metatags the admin setup not change and the search engine show what we like.

Here I post a example from Google where try to crawl the URL with the Variable How you can see I try to crawl and render the home page of blablanet.com but google got a URL REDIRECT from HUBZILLA HUB how you can see because the URL is not correct Metatags not show UP 

https://blablanet.com/	
URL submitted to index
Googlebot type: Desktop (render requested)
	Redirected on Monday, December 28, 2015 at 10:08:01 AM PST
This URL redirected to:	
https://blablanet.com/&JS=1

HTTP/1.1 200 OK
Server: cloudflare-nginx
Date: Mon, 28 Dec 2015 18:08:13 GMT
Content-Type: text/html; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Set-Cookie: __cfduid=d4fcb0beb166470e5382fad61d7ac9e1a1451326092; expires=Tue, 27-Dec-16 18:08:12 GMT; path=/; domain=.blablanet.com; HttpOnly
Set-Cookie: PHPSESSID=1lqkl0692737h7vrthjqf4abn2; path=/; HttpOnly
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache
Vary: Accept-Encoding
CF-RAY: 25bf35910ea10104-DFW

<!DOCTYPE html >
<html>
<head>
  <title>Blablanet Social Network</title>
  <script type="text/javascript">
//<![CDATA[
try{if (!window.CloudFlare) {var CloudFlare=[{verbose:0,p:1450973855,byc:0,owlid:"cf",bag2:1,mirage2:0,oracle:0,paths:{cloudflare:"/cdn-cgi/nexp/dok3v=1613a3a185/"},atok:"8c10ec1613c1d007cfaf8104824cfedc",petok:"3c160273a1b3a94c8918e0e9a9008dd4d1a487b4-1451326093-1800",zone:"blablanet.com",rocket:"0",apps:{"trumpet":{"message":"By using our website, you accept that we use cookies to perform analytics and produce content "}},sha2test:0}];!function(a,b){a=document.createElement("script"),b=document.getElementsByTagName("script")[0],a.async=!0,a.src="//ajax.cloudflare.com/cdn-cgi/nexp/dok3v=38857570ac/cloudflare.min.js",b.parentNode.insertBefore(a,b)}()}}catch(e){};
//]]>
</script>
<script>var baseurl="https://blablanet.com";</script>
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<base href="https://blablanet.com/" />
<meta name="viewport" content="width=device-width, height=device-height, initial-scale=1, user-scalable=1" />
<meta name="generator" content="hubzilla 2015-12-25.1257H" />



<link rel="stylesheet" href="https://blablanet.com/library/font_awesome/css/font-awesome.min.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/library/bootstrap/css/bootstrap.min.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/library/bootstrap-tagsinput/bootstrap-tagsinput.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/view/css/bootstrap-red.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/library/datetimepicker/jquery.datetimepicker.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/library/bootstrap-colorpicker/dist/css/bootstrap-colorpicker.min.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/library/tiptip/tipTip.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/library/jgrowl/jquery.jgrowl.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/library/jRange/jquery.range.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/view/css/conversation.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/view/css/widgets.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/view/css/colorbox.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/library/justifiedGallery/justifiedGallery.min.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/view/css/default.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/view/css/mod_home.css" type="text/css" media="screen">
<link rel="stylesheet" href="https://blablanet.com/view/theme/suckerberg/php/style.pcss" type="text/css" media="screen">


<script>

	var aStr = {

		'delitem'     : "Delete this item?",
		'comment'     : "Comment",
		'showmore'    : "[+] show all",
		'showfewer'   : "[-] show less",
		'divgrowmore' : "[+] expand",
		'divgrowless' : "[-] collapse",
		'pwshort'     : "Password too short",
		'pwnomatch'   : "Passwords do not match",
		'everybody'   : "everybody",
		'passphrase'  : "Secret Passphrase",
		'passhint'    : "Passphrase hint",
		'permschange' : "Notice: Permissions have changed but have not yet been submitted.",
		'closeAll'    : "close all",
		'nothingnew'  : "Nothing new here",
		'rating_desc' : "Rate This Channel (this is public)",
		'rating_val'  : "Rating",
		'rating_text' : "Describe (optional)",
		'submit'      : "Submit",
		'linkurl'     : "Please enter a link URL",
		'leavethispage' : "Unsaved changes. Are you sure you wish to leave this page?",
		'location'    : "Location",

		't01' : "",
		't02' : "",
		't03' : "ago",
		't04' : "from now",
		't05' : "less than a minute",
		't06' : "about a minute",
		't07' : "%d minutes",
		't08' : "about an hour",
		't09' : "about %d hours",
		't10' : "a day",
		't11' : "%d days",
		't12' : "about a month",
		't13' : "%d months",
		't14' : "about a year",
		't15' : "%d years",
		't16' : " ",
		't17' : "[]",

		'monthNames' : [ "January","February","March","April","May","June","July","August","September","October","November","December" ],
		'monthNamesShort' : [ "Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec" ],
		'dayNames' : ["Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"],
		'dayNamesShort' : ["Sun","Mon","Tue","Wed","Thu","Fri","Sat"],
		'today' : "today",
		'month' : "month",
		'week' : "week",
		'day' : "day",
		'allday' : "All day"
	};

</script>
		


<script src="https://blablanet.com/view/js/jquery.js" ></script>
<script src="https://blablanet.com/library/justifiedGallery/jquery.justifiedGallery.min.js" ></script>
<script src="https://blablanet.com/library/sprintf.js/dist/sprintf.min.js" ></script>
<script src="https://blablanet.com/view/js/spin.js" ></script>
<script src="https://blablanet.com/view/js/jquery.spin.js" ></script>
<script src="https://blablanet.com/view/js/jquery.textinputs.js" ></script>
<script src="https://blablanet.com/view/js/autocomplete.js" ></script>
<script src="https://blablanet.com/library/jquery-textcomplete/jquery.textcomplete.js" ></script>
<script src="https://blablanet.com/library/jquery.timeago.js" ></script>
<script src="https://blablanet.com/library/readmore.js/readmore.js" ></script>
<script src="https://blablanet.com/library/jgrowl/jquery.jgrowl_minimized.js" ></script>
<script src="https://blablanet.com/library/cryptojs/components/core-min.js" ></script>
<script src="https://blablanet.com/library/cryptojs/rollups/aes.js" ></script>
<script src="https://blablanet.com/library/cryptojs/rollups/rabbit.js" ></script>
<script src="https://blablanet.com/library/cryptojs/rollups/tripledes.js" ></script>
<script src="https://blablanet.com/view/js/acl.js" ></script>
<script src="https://blablanet.com/view/js/webtoolkit.base64.js" ></script>
<script src="https://blablanet.com/view/js/crypto.js" ></script>
<script src="https://blablanet.com/library/jRange/jquery.range.js" ></script>
<script src="https://blablanet.com/library/colorbox/jquery.colorbox-min.js" ></script>
<script src="https://blablanet.com/library/jquery.AreYouSure/jquery.are-you-sure.js" ></script>
<script src="https://blablanet.com/library/tableofcontents/jquery.toc.js" ></script>
<script src="https://blablanet.com/library/bootstrap/js/bootstrap.min.js" ></script>
<script src="https://blablanet.com/library/bootbox/bootbox.min.js" ></script>
<script src="https://blablanet.com/library/bootstrap-tagsinput/bootstrap-tagsinput.js" ></script>
<script src="https://blablanet.com/library/datetimepicker/jquery.datetimepicker.js" ></script>
<script src="https://blablanet.com/library/bootstrap-colorpicker/dist/js/bootstrap-colorpicker.js" ></script>
<script src="https://blablanet.com/view/theme/suckerberg/js/suckerberg.js" ></script>


<link rel="shortcut icon" href="https://blablanet.com/images/hz-32.png" />
<link rel="search"
         href="https://blablanet.com/opensearch" 
         type="application/opensearchdescription+xml" 
         title="Search in the Hubzilla" />


<script>

	var updateInterval = 80000;
	var localUser = false;
	var zid = null;
	var justifiedGalleryActive = false;
		
</script>




<meta name="robots" content="index" />
<meta name="description" content="BlaBlanet the social network service. Connect with friends, family . Share photos and videos, send messages and chat with people you know." />

<meta name="keywords" content="social, network, networking, service, friends, family, vpn, blogs, photos, videos, files, directory, encryption" />

<script>$(document).ready(function() {
	$("#nav-search-text").search_autocomplete('https://blablanet.com/acl');
});

</script><script src="https://blablanet.com/view/js/main.js" ></script>
</head>
<body>
	<header><div id="banner" class="hidden-sm hidden-xs">Blablanet</div>

<ul id="nav-notifications-template" style="display:none;" rel="template">
	<li class="{5}"><a href="{0}" title="{2} {3}"><span class="contactname">{2}</span><span class="dropdown-sub-text">{3}<br>{4}</span></a></li>
</ul>
</header>
	<nav class="navbar navbar-inverse navbar-fixed-top" role="navigation">	<div class="container-fluid">
		<div class="navbar-header">
			<button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#navbar-collapse-1">
				<span class="icon-bar"></span>
				<span class="icon-bar"></span>
				<span class="icon-bar"></span>
			</button>
			<button id="expand-tabs" type="button" class="navbar-toggle" data-toggle="collapse" data-target="#tabs-collapse-1">
				<i class="icon-circle-arrow-down" id="expand-tabs-icon"></i>
			</button>
			<button id="expand-aside" type="button" class="navbar-toggle" data-toggle="offcanvas" data-target="#region_1">
				<i class="icon-circle-arrow-right" id="expand-aside-icon"></i>
			</button>
					</div>
		<div class="collapse navbar-collapse" id="navbar-collapse-1">
			<ul class="nav navbar-nav navbar-left">
			<li class=""><a href="register" title="Create an account" id="register_nav_btn">Register</a></li>										<li class=" hidden-xs">
					<a data-toggle="dropdown" data-target="#" href="login" title="Sign in" id="login_nav_btn">Login <span class="caret" id="loginmenu-caret"></span></a>
					<ul class="dropdown-menu" role="menu">
												<li role="presentation"><a class="" href="login" title="Sign in" role="menuitem" id="login_nav_btn">Login</a></li>
												<li role="presentation"><a class="" href="rmagic" title="Click to authenticate to your home hub" role="menuitem" id="rmagic_nav_btn">Remote authentication</a></li>
											</ul>
				</li>
								<li role="presentation"><a class=" visible-xs" href="login" title="Sign in" role="menuitem">Login</a></li>
								<li role="presentation"><a class=" visible-xs" href="rmagic" title="Click to authenticate to your home hub" role="menuitem">Remote authentication</a></li>
										
			
			
			

			
			
					
						</ul>
			<ul class="nav navbar-nav navbar-right">
				<li class="hidden-xs">
					<form method="get" action="search" role="search">
						<div id="nav-search-spinner"></div><input class="icon-search" id="nav-search-text" type="text" value="" placeholder="&#xf002; @name, #tag, ?doc, content" name="search" title="Search site @name, #tag, ?docs, content" onclick="this.submit();"/>
					</form>
				</li>
				<li class="visible-xs">
					<a href="/search" title="Search site @name, #tag, ?docs, content"><i class="icon-search"></i></a>
				</li>

										<li class="">
					<a class="" href="directory" title="Channel Directory" id="directory_nav_btn"><i class="icon-sitemap"></i></a>
				</li>
			
							<li class="">
					<a class="" href="apps" title="Applications, utilities, links, games" id="apps_nav_btn"><i class="icon-cogs"></i></a>
				</li>
			
							<li class="">
					<a class="" target="hubzilla-help" href="https://blablanet.com/help?f=&cmd=" title="Help and documentation" id="help_nav_btn"><i class="icon-question"></i></a>
				</li>
						</ul>
		</div>
	</div>
</nav>
	<main>
		<aside id="region_1"></aside>
		<section id="region_2"><script>document.cookie="jsAvailable=1; path=/"; var jsMatch = /\&JS=1/; if (!jsMatch.exec(location.href)) { location.href = location.href + "&JS=1"; }</script><h1 class="home-welcome">Welcome to Blablanet Social Network</h1><div class="generic-content-wrapper-styled">
<form action="https://blablanet.com/" id="main-login" method="post" >
	<input type="hidden" name="auth-params" value="login" />
	<div id="login-main">
		<div id="login-input" class="form-group">
				<div class='form-group field input'>
		<label for='id_username' id='label_username'>Email</label>
		<input class="form-control" name='username' id='id_username' type="text" value="">
		<span id='help_username' class='help-block'></span>
		<div class="clear"></div>
	</div>

				<div class='form-group field password'>
		<label for='id_password'>Password</label>
		<input class="form-control" type='password' name='password' id='id_password' value="">		<span class='help-block'></span>
	</div>

				<div id="remember_container" class="form-group field checkbox">
		<label for="id_remember">Remember me</label>
		<div class="pull-right"><input type="checkbox" name='remember' id='id_remember' value="1"   /><label class="switchlabel" for='id_remember'> <span class="onoffswitch-inner" data-on='Yes' data-off='No'></span><span class="onoffswitch-switch"></span></label></div>
		<span class="help-block"></span>
	</div>
	<div class="clear"></div>

			<button type="submit" name="submit" class="btn btn-block btn-primary">Login</button>
		</div>
		<div id="login-extra-links">
			<a href="register" title="Create an account to access services and applications within the Hubzilla" id="register-link" class="pull-right">Register</a>			<a href="lostpass" title="Forgot your password?" id="lost-password-link" >Password Reset</a>
		</div>
	</div>
			<input type="hidden" name="0" value="" />
	</form>
</div>
<script type="text/javascript"> $(document).ready(function() { $("#id_username").focus();} );</script>
					<div id="page-footer"></div>
			<div id="pause"></div>
		</section>
		<aside id="region_3" class="hidden-sm hidden-xs hidden-md"></aside>
	</main>
	<footer></footer>
</body>
</html>

**Comment by redmatrix** [comment]
I'm merely trying to provide information allowing you to work *with* Hubzilla code rather than create ugly hacks to work *around* it. You should never have to deal with the string '&JS=1'. We provide higher level tools to work with.  If we ever change the name of that variable your code will break. If you follow my advice, you don't even need to concern yourself with this variable and your code will work perfectly. It's up to you. Do whatever you want.
